macOS/DuckDuckGo/Updates/Sparkle/SparkleUpdateController.swift

```diff
@@ -145,12 +145,6 @@ final class SparkleUpdateController: NSObject, SparkleUpdateControllerProtocol {
             if newValue != areAutomaticUpdatesEnabled {
                 updateWideEvent.cancelFlow(reason: .settingsChanged)
                 userDriver?.cancelAndDismissCurrentUpdate()
-
-                if useLegacyAutoRestartLogic {
-                    updater = nil
-                } else {
-                    updater?.resetUpdateCycle()
-                }
```

Comment:
Additionally instead of setting `updater = nil` I'm adjusting the code so that `configureUpdater()` below will always just recreate the updater.  This helps us prevent doing this in two steps separated by an async delayed call, which can be a good recipe for race conditions (that I suspect may be at play here).

Regarding the removal of `updater?.resetUpdateCycle()`, this was added for the new logic, but it doesn't do what we need it to do - this won't affect current UX as it's disabled.